backend/app/services/employee_emergency_contact_service.py

```
@@ -444,14 +456,16 @@ async def delete_emergency_contact(
         # Authorization check
-        user_permissions = await get_user_permissions(current_user.user_id, db)
+        user_permissions = await get_user_permissions(user_id, db)
         if not any(p == Permission.DELETE_EMERGENCY_CONTACT.value or p == Permission.MANAGE_EMPLOYEES.value for p in user_permissions):
             raise HTTPException(
                 status_code=status.HTTP_403_FORBIDDEN,
                 detail="Not authorized to delete this emergency contact"
             )
 
-        old_values = db_contact.__dict__.copy()
+        # Serialize old values properly before deletion
+        old_values_dict = EmployeeEmergencyContactOut.model_validate(db_contact).model_dump(mode='json')
```

Ruff (0.12.2)

Local variable `old_values_dict` is assigned to but never used

Remove assignment to unused variable `old_values_dict`

(F841)